src/test/java/org/opensearch/knn/index/KNNMethodContextTests.java

```diff
@@ -144,6 +147,50 @@ public void testRequiresTraining() {
         assertTrue(knnMethodContext.isTrainingRequired());
     }
 
+    public void testEstimateOverheadInKB() {
+        // For HNSW no encoding we expect 0
+        MethodComponentContext hnswMethod = new MethodComponentContext(METHOD_HNSW, Collections.emptyMap());
+        KNNMethodContext knnMethodContextNmslib = new KNNMethodContext(KNNEngine.NMSLIB, SpaceType.L2, hnswMethod);
+        KNNMethodContext knnMethodContextFaiss = new KNNMethodContext(KNNEngine.FAISS, SpaceType.INNER_PRODUCT, hnswMethod);
+        assertEquals(0, knnMethodContextNmslib.estimateOverheadInKB(1000));
+        assertEquals(0, knnMethodContextFaiss.estimateOverheadInKB(168));
+
+        // For IVF, we expect  4 * nlist * d / 1024 + 1
+        int dimension = 768;
+        int nlists = 1024;
+        long expected = 4 * nlists * dimension / BYTES_PER_KILOBYTES + 1;
+
+        MethodComponentContext ivfMethod = new MethodComponentContext(METHOD_IVF, ImmutableMap.of(
+            METHOD_PARAMETER_NLIST, nlists
+        ));
+        knnMethodContextFaiss = new KNNMethodContext(KNNEngine.FAISS, SpaceType.L2, ivfMethod);
+        assertEquals(expected, knnMethodContextFaiss.estimateOverheadInKB(dimension));
+
+        // For IVFPQ twe expect  4 * nlist * d / 1024 + 1 + 4 * d * 2^code_size / 1024 + 1
+        int codeSize = 16;
+        long expectedFromPq = 4 * dimension * (1 << codeSize) / BYTES_PER_KILOBYTES + 1;
+        expected = expected + expectedFromPq;
```

Comment:
shall we keep new variable name instead of overriding it? During debug we will lose original value.